lib/streamlit/elements/image.py

```diff
@@ -355,6 +356,85 @@ def image_to_url(
     return this_file.url
 
 
+def _normalize_xmlns_attribute(root: ET.Element) -> None:
+    if "xmlns" not in root.attrib:
+        root.set("xmlns", "http://www.w3.org/2000/svg")
+
+
+def _parse_width_from_width_attribute(root: ET.Element) -> int:
+    width = 0
+    if "width" in root.attrib:
+        try:
+            # Get the width value based on width attribute
+            width = int(root.attrib.get("width", "").replace("px", ""))
+        except (TypeError, ValueError):
+            pass
+    return width
+
+
+def _parse_width_from_viewbox_attribute(root: ET.Element) -> int:
```

Comment:
I'm not sure if we actually can use the viewbox `width` here as well since it defines the defines the region of the SVG that will be visible and not the actual with. But I'm not too much into SVGs, maybe it is a good fallback.

Reply:
Seems like with this code the cases from [Issue-#3724](https://github.com/streamlit/streamlit/issues/3724) are resolved, with a fallback when exception occurs.
I've updated this stale PR, if you can please take a look at the examples from [Issue-#3724](https://github.com/streamlit/streamlit/issues/3724), code for them is below:
```
import streamlit

x = 50
streamlit.image(f"""<svg viewBox="{x} 0 100 90" xmlns="http://www.w3.org/2000/svg">
    <rect x="0" y="0" width="100" height="90" fill="yellow" />
    <rect x="100" y="0" width="100" height="90" fill="green" />
</svg>
""", width=100)
streamlit.image(f"""<svg viewBox="{x} 0 100 90" xmlns="http://www.w3.org/2000/svg">
    <rect x="0" y="0" width="100" height="90" fill="yellow" />
    <rect x="100" y="0" width="100" height="90" fill="green" />
</svg>
""", width=1000)

circle = """
<svg height="100" width="100">
  <circle cx="50" cy="50" r="40" stroke="black" stroke-width="3" fill="red" />
  Sorry, your browser does not support inline SVG.
</svg>
"""
streamlit.write("st.image(circle, width=100)")
streamlit.image(circle, width=100)
streamlit.write("st.image(circle, width=1000)")
streamlit.image(circle, width=1000)
```
I wonder what kind of tests we can perform here? Screenshot tests?